hooks/tk-multi-publish2/nuke/publish_comp.py

```diff
@@ -166,7 +126,7 @@ def validate(self, task_settings, item):
             status = self._bbsize(item) and status
             if item.properties['fields'].get('output') == 'main':
                 status = self._sync_frame_range(item) and status
-            status = self._framerange_to_be_published(item) and status
+                status = self._framerange_to_be_published(item, log_method="error") and status
```

Comment:
Since we are calling super in line 134 and this is also calling `_framerange_to_be_published()` in publish.py, maybe we can remove this? Or we can remove it from publish.py and move it to subclasses, since this `main` segregation is only needed for comp.

Reply:
Moving it to subclasses would mean that we would need to create a separate hook for each nuke based role for`_framerange_to_be_published()` or any other validations that may behave in such a manner i.e warning as default but error for a specific role or a specific case.

@garvitverma suggested using item.properties to set log_method to "error" when required and the validations can log "error" or "warning" based on this log_method. This way we are calling each validation only once.

Reply:
That works.